megamek/src/megamek/common/TankTrailerHitch.java

```diff
@@ -0,0 +1,291 @@
+
+    /**
+     * 
+     */
+    private static final long serialVersionUID = 1193349063084937973L;
+    
+    /**
+     * The entity being towed by this hitch.
+     */
+    protected int towed = Entity.NONE;
+    transient IGame game;
+
+    /**
+     * The front location that loads trailers externally.
+     */
+    private static final int EXTERIOR_LOCATIONS_FRONT = Tank.LOC_FRONT;
+
+    /**
+     * The rear location that loads trailers externally.
+     */
+    private static final int EXTERIOR_LOCATIONS_REAR = Tank.LOC_REAR;
+
+    /**
+     * The <code>String</code> reported when the hitch is in use.
+     */
+    private static final String NO_VACANCY_STRING = "A trailer is attached";
+
+    /**
+     * The <code>String</code> reported when the hitch is available.
+     */
+    private static final String HAVE_VACANCY_STRING = "One trailer";
+
+    // Protected constructors and methods.
+
+    /**
+     * Get the exterior locations that a trailer covers.
+     * <p/>
+     * Sub-classes are encouraged to override this method.
+     *
+     * @param isRear
+     *            - a <code>boolean</code> value stating if the given location
+     *            is rear facing; if <code>false</code>, the location is front
+     *            facing.
+     * @return an array of <code>int</code> listing the exterior locations.
+     */
+    protected int getExteriorLocs(boolean isRear) {
+        if (isRear) {
+            return TankTrailerHitch.EXTERIOR_LOCATIONS_REAR;
+        }
+        return TankTrailerHitch.EXTERIOR_LOCATIONS_FRONT;
+    }
+
+    /**
+     * Get the <code>String</code> to report the presence (or lack thereof) of a
+     * towed trailer.
+     * <p/>
+     * Sub-classes are encouraged to override this method.
+     *
+     * @param isLoaded
+     *            - a <code>boolean</code> that indicates a trailer is
+     *            currently loaded (if the value is <code>true</code>) or not
+     *            (if the value is <code>false</code>).
+     * @return a <code>String</code> describing the occupancy state of this
+     *         transporter.
+     */
+    protected String getVacancyString(boolean isLoaded) {
+        if (isLoaded) {
+            return TankTrailerHitch.NO_VACANCY_STRING;
+        }
+        return TankTrailerHitch.HAVE_VACANCY_STRING;
+    }
+
+    // Public constructors and methods.
+
+    /**
+     * Create a new hitch.
+     */
+    public TankTrailerHitch() {
+    }
+
+    /**
+     * Determines if this object can accept the given unit. The unit may not be
+     * of the appropriate type or there may be no room for the unit.
+     * <p>
+     *
+     * @param unit
+     *            - the <code>Entity</code> to be loaded.
+     * @return <code>true</code> if the unit can be loaded, <code>false</code>
+     *         otherwise.
+     */
+    public boolean canLoad(Entity unit) {
+        // Only trailers can be towed.
+        if (!(unit.isTrailer())) {
```

Comment:
Remove superfluous parentheses.

Reply:
Fixed. This comes from some problems I've had where my ! doesn't apply correctly without the parentheses.